encodings/fastlanes/src/for/compare.rs

```diff
@@ -0,0 +1,44 @@
+use vortex_array::compute::{compare, CompareFn, Operator};
+use vortex_array::ArrayData;
+use vortex_error::VortexResult;
+
+use crate::{decompress, FoRArray, FoREncoding};
+
+impl CompareFn<FoRArray> for FoREncoding {
+    fn compare(
+        &self,
+        lhs: &FoRArray,
+        rhs: &ArrayData,
+        operator: Operator,
+    ) -> VortexResult<Option<ArrayData>> {
+        // this is cheap
+        let owned_lhs = lhs.clone();
+        let decompressed_lhs = decompress(owned_lhs)?;
```

Comment:
So by default, the compare function will fall back to canonicalizing an array into the Arrow representation and then running the compare kernel over it.

So we should only implement a compute function for an encoding when it is able to short-cut some work.

In the case of FoR, if the RHS is constant `rhs.as_constant()`, then you could subtract the `lhs.reference()` value and then push-down the comparison without decompressing. 

For FoR, this doesn't itself make the compare any cheaper, but it does allow FoR's child to possible implement a cheaper comparison.